docs/_blog/bdk_core_pt1.md

```diff
@@ -0,0 +1,322 @@
+5. Given some parameters it can build a PSBT from transaction outputs.
+6. Given a PSBT it can sign it with its [`Signers`][`Signer`].
+
+All of that is very useful but it is bound together with the particular policies and opinions of `Wallet`.
+If `Wallet`'s policy is not your policy it's going to be tricky to get it to do what you want.
+Here's some examples:
+
+1. In order to control how the `Wallet` will select coins for a transaction internally you have to
+   pass in something implementing the [`CoinSelectionAlgorithm`] trait. A coin selection algorithm
+   is clearly mechanism code but the policy of `Wallet` restricts that mechanism's interface. We
+   have [very old issues](https://github.com/bitcoindevkit/bdk/issues/281) related to what the
+   interface of this trait should be and we don't have a clear way forward. In `bdk_core` I want to
+   purely provide the coin selection mechanisms for figuring out whether you need to select more
+   UTXOs or whether you need a change output etc. How you use that mechanism will be up to you.
+2. Another trait that has a similar structure is the [`Signer`] trait. You have to pass in signers
+   so your wallet can sign PSBTs but you have little control on how the wallet chooses which signers
+   to use any given situation. Right now the wallet will just iterate through all the signers and
+   ask them to sign. This is not always the appropriate. In `bdk_core` I hope we can just provide
+   ways of populating the correct PSBT fields with signatures.
+
+## A Syncing mechansim without the policy
+
+The rest of this post goes into detail about how to expose useful mechanisms for syncing without
+imposing a policy either on how blockchain data is fetched or how it is stored.
+
+### The problems with the policy
+
+Syncing in `bdk` is the place where the design of `Wallet` is most restrictive. The [`WalletSync`]
+trait forces you to sync all addresses in a wallet in one big batch. But this is not always what you
+want to do. I spoke to a developer who wanted to create a new tor connection to an electrum server
+for each address so the addresses couldn't be easily linked and to run this slowly in the
+background. It would be really difficult to to implement `WalletSync` with such a strategy. Another
+example where `WalletSync` isn't the right fit is the [Sensei] project which uses BDK but
+incrementally updates the database whenever new information comes in from the blockchain.
+
+Even if syncing all addresses at the same time is roughly what you want to do `WalletSync` still
+gets in the way since it defines whether you do it synchronously or asynchrononusly. Applications
+can control this through `bdk`'s `async-interface` feature flag which internally changes the trait
+definition through macros. Another annoyance is that when using `async-interface` the future that
+gets returned from `WalletSync` [cannot be `Send`](https://github.com/bitcoindevkit/bdk/issues/165)
```

Comment:
[cannot be sent (`Send`)]

Maybe?

Reply:
cannot be `Send` implies it cannot be *sent* across threads. I think the sentence is ok the way it is.